encoding/graph/definition/definition_key_generator.rs

```diff
@@ -28,4 +32,13 @@ impl DefinitionKeyGenerator {
         snapshot.put(definition_key.as_storage_key().into_owned_array());
         Ok(definition_key)
     }
+
+    pub fn create_function<Snapshot: WritableSnapshot>(
+        &self,
+        snapshot: &mut Snapshot,
+    ) -> Result<DefinitionKey<'static>, EncodingError> {
+        let definition_key = self.next_function.allocate(snapshot)?;
+        snapshot.put(definition_key.as_storage_key().into_owned_array());
+        Ok(definition_key)
+    }
```

Comment:
Do we want to split this into `StructTypeIDGenerator` and `FunctionIDGenerator`?

Reply:
It's ok like this